### DISCUSSION TITLE
Mock websocket server for unit tests

### DISCUSSION THREAD

**Discussion by JulienChampagnol** [nuxt/test-utils#764] [Q&A]
### Describe the feature

The same way registerEndpoint mocks a server endpoint, 
is it possible to create an util that mocks a websocket server?

### Additional information

- [ ] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by danielroe** [comment]
Are you wanting to test Nuxt HMR or are you doing something else with websockets?

**Comment by JulienChampagnol** [comment]
The goal is to test to mock external features inside of tests

**Comment by danielroe** [comment]
So this doesn't really seem like a nuxt/test-utils feature request so much as general testing advice, right? You can of course create a websocket server in a test. But I think I (or anyone else) might need some more info on your app, what you are trying and what isn't working for you.
